src/MyStack/MyLinkedList.cs

```diff
@@ -56,9 +57,35 @@ public void AddFirst(T data)
         Count++;
     }
 
-    public void Remove()
+    public void Remove(T value)
     {
+        if (Count == 0)
+        {
```

Comment:
Нужно ли здесь это исключение? Если нужно, то аналогичное исключение нужно генерировать если не найден искомый элемент для удаления
Либо вообще отказаться от исключений и спроектировать метод Remove так, чтобы он возвращал true или false (удаление успешно или нет)

Reply:
Я исходил из логики, что пользователь удаляет какое-то число, но если этого числа нет, то программа не будет делать ничего

Reply:
Да и мне кажется, что получать из метода ответ о удалении не очень удобно. Пользователь скорее всего просто захочет удалить переменную не отслеживая это действие дополнительно. В таком случае лучше было бы добавить ивент, на который можно было бы подписаться, если пользователю очень захочется следить за ходом удаления элементов коллекции

Reply:
> Я исходил из логики, что пользователь удаляет какое-то число, но если этого числа нет, то программа не будет делать ничего

по поводу этой логики я согласен. но ведь сейчас эта логика не соблюдается для случая, когда список пуст?
КАК ДОЛЖНО БЫТЬ: список пуст -> искомого элемента в списке нет -> метод ничего не делает
КАК ЕСТЬ: список пуст -> искомого элемента в списке нет -> метод генерирует исключение и возникает ошибка

по поводу возврата bool из метода. в целом, принято реализовывать метод remove, который возвращает bool. например, давай посмотрим [документацию](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.remove?view=net-8.0) метода Remove коллекции List<T>. по поводу возвращаемого значения здесь написано:

> Returns
Boolean
true if item is successfully removed; otherwise, false. This method also returns false if item was not found in the List<T>.